File: frontend/recruitment_platform_app/lib/models/create_job_request.dart
class CreateJobRequest {
  final String title;
  final String description;
  final String requirements;
  final String location;
  final String workType;

  CreateJobRequest({
    required this.title,
    required this.description,
    required this.requirements,
    required this.location,
    required this.workType,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'requirements': requirements,
      'location': location,
      'workType': workType,
    };
  }
}

File: backend/job-service/src/main/java/com/recruitment/platform/job/dto/CreateJobRequest.java
package com.recruitment.platform.job.dto;

// DTO for creating a new job posting
public record CreateJobRequest(
    String title,
    String description,
    String requirements,
    String salaryRange,
    String benefits,
    String location,
    String workType,
    Long positionId,
    Integer hiringQuantity
    // recruiterId will be set from JWT, companyId from header
) {}
